packages/caliper-core/test/common/core/transaction-statisitcs-collector.js

```diff
@@ -0,0 +1,219 @@
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            clock.tick(50);
+            txStatus.SetStatusSuccess();
+
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            transactionStatisticsCollector.getTotalFinishedTx().should.equal(1);
+            transactionStatisticsCollector.getTotalSuccessfulTx().should.equal(1);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
+            transactionStatisticsCollector.getLastCreateTime().should.equal(50);
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(100);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(100);
+            transactionStatisticsCollector.getMinLatencyForSuccessful().should.equal(50);
+            transactionStatisticsCollector.getMaxLatencyForSuccessful().should.equal(50);
+            transactionStatisticsCollector.getTotalLatencyForSuccessful().should.equal(50);
+        });
+
+        it('should update failed transactions', () => {
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus = new TxStatus();
+            clock.tick(50);
+            txStatus.SetStatusFail();
+
+            transactionStatisticsCollector.txFinished(txStatus);
+
+            transactionStatisticsCollector.getTotalFinishedTx().should.equal(1);
+            transactionStatisticsCollector.getTotalFailedTx().should.equal(1);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
+            transactionStatisticsCollector.getLastCreateTime().should.equal(50);
+            transactionStatisticsCollector.getFirstFinishTime().should.equal(100);
+            transactionStatisticsCollector.getLastFinishTime().should.equal(100);
+            transactionStatisticsCollector.getMinLatencyForFailed().should.equal(50);
+            transactionStatisticsCollector.getMaxLatencyForFailed().should.equal(50);
+            transactionStatisticsCollector.getTotalLatencyForFailed().should.equal(50);
+        });
+
+        it('should be possible update with an array of transactions', () => {
+
+            const transactionStatisticsCollector = new TransactionStatisticsCollector();
+            transactionStatisticsCollector.activate();
+
+            clock.tick(50);
+            const txStatus0 = new TxStatus();
+            clock.tick(25);
+            const txStatus1 = new TxStatus();
+            txStatus0.SetStatusSuccess();
+            clock.tick(100);
+            txStatus1.SetStatusSuccess();
+
+            transactionStatisticsCollector.txFinished([txStatus0, txStatus1]);
+            transactionStatisticsCollector.getTotalSuccessfulTx().should.equal(2);
+            transactionStatisticsCollector.getFirstCreateTime().should.equal(50);
```

Comment:
It might be better to use the `txStatus.GetXY()` return values for the expected values, rather than literal numbers. That will clearly indicate the source of the expected value. (Only applies if a metric is derived from txStat metrics.)